R/boxr_delete_restore.R

```diff
@@ -91,7 +91,8 @@ boxDeleteFile <- function(file_id) {
       "https://api.box.com/2.0/files/",
       file_id
     ),
-    httr::config(token = getOption("boxr.token"))
+    httr::config(token = getOption("boxr.token")),
```

Comment:
It might be worth having an internal function `get_token()`, something like:

```r
get_token <- function() {
  httr::config(token = getOption("boxr.token")) %||% getOption("boxr_token_jwt")
}
```

Reply:
right now the JWT is being carried in a Bearer header via `httr::add_headers()` to play nicely with [Box's API](https://developer.box.com/docs/construct-jwt-claim-manually#section-3-create-jwt-assertion) . I don't believe `httr::config(token = )` is compatible here

Reply:
something like this done in 4fb05f52a74614fe980f740ea2a81498363c8b53